examples/src/java/org/pantsbuild/example/3rdparty_jvm.md

```diff
@@ -291,3 +291,240 @@ time pants is run (otherwise, pants will cache it):
+
+    :::python
+    scala_library(
+        name='A',
+        source=['A.scala'],
+    )
+
+    scala_library(
+        name='B',
+        source=['B.scala'],
+        dependencies=[':A'],
+    )
+
+    scala_library(
+        name='C',
+        source=['C.scala'],
+        dependencies=[':B'],
+    )
+
+If we compile C with strict_deps turned off, everything works fine, as all
+transitive dependencies of C will be added to compile time classpaths.
+However, the compilation may require additional dependencies if strict_deps
+is turned on. With strict_deps=True, only B is in C's compile time classpath.
+There are 2 cases for the failure.
+
+1. C directly uses APIs from A. This is obviously a mistake by author of C,
+and the fix is simple, just add A to C's dependency list.
+
+2. C does not use APIs from A, but compiler still wants A to be in C's
+classpaths. This is tricky, as author of C may not even be aware of A. It's
+hard for C's author to get the right dependency list in this case.
+
+People may wonder why the compiler asks for A to compile C in the first place.
+There are a number of reasons for this, an example of which is that the java
+compiler needs all transitively implemented interfaces to be on the classpath
+so that it can resolve default method implementations.
+
+### Strict_deps for Library Developers
+
+The compile overhead of a library change for downstream targets when
+strict_deps is enabled should be minimized to just the targets that
+depend directly on the library target or depend on a target that exports
+that target.
+
+#### Invalidation
+
+This is a change from the current behavior where all downstream targets will
+have their compile invalidated.
+
+With strict_deps, a change to a library target will only invalidate the
+compile artifacts of dependent targets if they have
+
+1. A direct dependency on the changed library target, or
+2. A dependency on it via the export graph.
+
+Before, if a target depending on a library had an implicit dependency on a
+transitive dependency of the library target it would just work. The transitive
+dependency would end up on the dependent target’s compile classpath and that
+target would compile happily. Strict deps removes that implicit transitivity,
+which exposes undeclared dependencies.
+
+This introduces two new sources of errors.
+
+1. When adding a new library dependency to a target, if the library dependency
+requires some of its dependencies to be on the compile classpath, the compile
+can fail if those dependencies don’t make it on the classpath. These errors
+are a bit weird and will be confusing at first.
+2. When a library target adds a new publicly visible dependency, targets that
+depend on the library target will fail to compile with the same kind of error
+message as 1.
+
+### Errors, Exports, and Missing-deps-suggest
+
+The errors sometimes have different messages, but usually in Scala they look
+something like this:
+
+    :::bash
```

Comment:
`:::bash` shows up in the doc. same for others. I think the syntax should be https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting